apps/api/src/tasks/bank-import/import-transactions.task.ts

```diff
@@ -298,8 +307,23 @@ export class IrisTasks {
         .replace(/[ _]+/g, '-')
         .match(this.regexPaymentRef)
 
+      const transactionAmount = {
+        amount: trx.transactionAmount?.amount,
+        currency: trx.transactionAmount?.currency,
+      }
+      const id = trx.transactionId?.trim() || ''
+
+      // If we receive EUR in the BGN account - try parsing from the transaction id the amount in BGN
+      if (trx.transactionAmount?.currency === Currency.EUR && trx.transactionAmount?.amount > 0) {
+        const amount = this.extractAmountFromTransactionId(id, trx.valueDate)
+        if (amount) {
+          transactionAmount.amount = amount
+          transactionAmount.currency = Currency.BGN
+        }
```

Comment:
Currently the "logic" is - if we can parse the Amount and it's a positive one (> 0) then we set update the transaction. Otherwise, we won't touch it and it will be recorded as received - therefore not transformed and we can see what's the case and how to handle it. It's kind of defensive - if the code is not sure, it doesn't change anything.

Again, that's just my concern here.